src/injected/analyzers/tab-stops-orchestrator.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { WindowUtils } from 'common/window-utils';
+import { AllFrameRunnerTarget } from 'injected/all-frame-runner';
+import {
+    TabStopRequirementResult,
+    TabStopsRequirementEvaluator,
+} from 'injected/tab-stops-requirement-evaluator';
+import { TabbableElementGetter } from 'injected/tabbable-element-getter';
+import { FocusableElement } from 'tabbable';
+
+export class TabStopRequirementOrchestrator
+    implements AllFrameRunnerTarget<TabStopRequirementResult>
+{
+    public readonly name: string = 'TabStopRequirementOrchestrator';
+    private tabbableTabStops: FocusableElement[];
+    private actualTabStops: Set<HTMLElement> = new Set();
+    private latestVisitedTabStop: HTMLElement = null;
+    private reportResults: (payload: TabStopRequirementResult) => Promise<void>;
+
+    constructor(
+        private readonly dom: Document,
+        private readonly tabbableElementGetter: TabbableElementGetter,
+        private readonly windowUtils: WindowUtils,
+        private readonly tabStopsRequirementEvaluator: TabStopsRequirementEvaluator,
+        private readonly getUniqueSelector: (element: HTMLElement) => string,
+    ) {}
+
+    public start = () => {
+        this.dom.addEventListener('keydown', this.onKeydownForFocusTraps);
+        this.dom.addEventListener('focusin', this.addNewTabStop);
+        this.tabbableTabStops = this.tabbableElementGetter.getRawElements();
+        const tabbableFocusOrderResults =
+            this.tabStopsRequirementEvaluator.getTabbableFocusOrderResults(this.tabbableTabStops);
+        tabbableFocusOrderResults.forEach(result => {
+            this.reportResults(result);
```

Comment:
I think it's probably possible that we'll have duplicated results between `getTabbableFocusOrderResults` and `getFocusOrderResult`. Are we okay with reporting those duplicates or do we need to add logic to filter those out?

Reply:
Thought about that as well and figured it might make more sense to do later, in the process that is.